pkg/sessions/tests/mock_store.go

```diff
@@ -0,0 +1,43 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+)
+
+// Store is a generic in-memory implementation of persistence.Store
+// for mocking in tests
+type Store struct {
+	cache *cache.Cache
```

Comment:
What's the benefit of using a cache here over a map? Can we fast forward the cache somehow to pretend time has passed (eg like miniredis does)?

Reply:
I was actually curious when I first wrote this last night -- I was looking to see if there was any support for that in cache (there wasn't) -- and then assumed when I ran the test suite, would I fail tests because I made the fast-forward func nil.

I surely thought I would have expired entry issues not being expired fail tests... but none failed 😢 

I'll look into this more to see if something is missing in that test logic.

Broader question for future stores: a `persistentFastForwardFunc` handler might not be as readily available in other DB mocking libraries for new stores like it is in miniredis. We don't want to make plugging in new DB stores into the `Manager` framework seamless only to making testing a nightmare since they can't easily simulate time fast forwards 🤔

Reply:
Oh lol... likely because it doesn't even run them when that handler is nil:
```
			if persistentFastForward != nil {
				PersistentSessionStoreInterfaceTests(&input)
			}
```

Reply:
Made our own that can fast forwards 👍 

You were right, using the cache library really didn't save many any complexity in implementation details vs just making our own Map.

Reply:
> Oh lol... likely because it doesn't even run them when that handler is nil:

Yep, that was very deliberate, there's a subset of the tests we just can't run if we don't have a fast forward. So when mocking the session stores, we should endeavor to mock the fast forward somehow

> You were right, using the cache library really didn't save many any complexity in implementation details vs just making our own Map.

Great! Will take a look